annotation-processor/src/main/java/com/linecorp/armeria/server/annotation/processor/DocumentationProcessor.java

```diff
@@ -0,0 +1,186 @@
+            properties.store(writer, "Generated list of parameter description of REST interfaces.");
+        }
+    }
+
+    private void processAnnotation(TypeElement annotationElement, RoundEnvironment roundEnv) {
+        roundEnv.getElementsAnnotatedWith(annotationElement)
+                .stream()
+                .filter(element -> element.getKind() == ElementKind.METHOD)
+                // Element is always ExecutableElement because it is a method.
+                .forEachOrdered(element -> {
+                    try {
+                        processMethod((ExecutableElement) element);
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+    }
+
+    private void processMethod(ExecutableElement method) throws IOException {
+        final String className = ((TypeElement) method.getEnclosingElement()).getQualifiedName().toString();
+        final Properties properties = readProperties(className);
+        final String docComment = processingEnv.getElementUtils().getDocComment(method);
+        if (docComment == null || !docComment.contains("@param")) {
+            return;
+        }
+        final List<List<String>> lines = Arrays.stream(docComment.split("\\R"))
+                                               .filter(line -> !line.trim().isEmpty())
+                                               .map(line -> Arrays.stream(line.split("\\s"))
+                                                                  .filter(word -> !word.trim().isEmpty())
+                                                                  .collect(toImmutableList()))
+                                               .collect(toImmutableList());
+        method.getParameters().forEach(param -> {
+            final StringBuilder stringBuilder = new StringBuilder();
+            JavaDocParserState state = JavaDocParserState.SEARCHING;
+            for (List<String> line : lines) {
+                final List<String> subLine;
+                if ((line.size() < 3 ||
+                     !"@param".equals(line.get(0)) ||
+                     !param.getSimpleName().toString().equals(line.get(1))) &&
+                    state == JavaDocParserState.SEARCHING) {
+                    continue;
+                } else if (state == JavaDocParserState.IN_DESCRIPTION &&
+                           line.size() > 0 &&
+                           "@param".equals(line.get(0))) {
+                    break;
+                } else if (state == JavaDocParserState.SEARCHING) {
+                    subLine = line.subList(2, line.size());
+                    state = JavaDocParserState.IN_DESCRIPTION;
+                } else {
+                    subLine = line;
+                }
+                for (String word : subLine) {
+                    stringBuilder.append(word);
+                    stringBuilder.append(' ');
+                }
+            }
```

Comment:
1) This will not cover a Javadoc which a user adds another tags such as `@returns`, `@throws` to. 
2) If use switch statement, we can simplify this a little bit.

```java
outer:
for (List<String> line : lines) {
    switch (state) {
        case SEARCHING:
            final int size = line.size();
            if (size < 3 ||
                !"@param".equals(line.get(0)) ||
                !param.getSimpleName().toString().equals(line.get(1))) {
                break;
            }
            line = line.subList(2, size);
            state = JavaDocParserState.IN_PARAM;
        case IN_PARAM:
            if (!line.isEmpty()) {
                final String first = line.get(0);
                // break loop if first is `@xxxxx`
                if (!first.isEmpty() && first.charAt(0) == '@') {
                    break outer;
                }

                for (String word : line) {
                    stringBuilder.append(word);
                    stringBuilder.append(' ');
                }
            }
            break;
        }
}
```

Reply:
I disagree that `2` is simpler than my approach. I dislike the labeled break... 

I've added tests for `@throws` and `@return` and added handling for them...

Reply:
I think simplicity is not a problem if it can produce the same result. 
If you don't like the labeled break, that sounds good to me.